test/ReadsUtils_server_test.py

```diff
@@ -3063,18 +3085,16 @@ def test_upload_reads_from_web_ftp_anonymous(self):
         fq_path = os.path.join(self.cfg['scratch'], fq_filename)
         shutil.copy(os.path.join("data", fq_filename), fq_path)
 
-        ftp_connection = ftplib.FTP('ftp.uconn.edu')
-        ftp_connection.login('anonymous', 'anonymous@domain.com')
-        ftp_connection.cwd("/48_hour/")
-
-        if fq_filename not in ftp_connection.nlst():
-            fh = open(os.path.join("data", fq_filename), 'rb')
-            ftp_connection.storbinary('STOR Sample1.fastq', fh)
-            fh.close()
+        with ftplib.FTP(self.ftp_domain) as ftp_connection:
```

Comment:
Consider making this a function, it's duplicated several times in the tests

Reply:
Stylistic - boilerplate is wasteful, but in test code is is not as awful. Nice to have, but not a hard requirement.